### DISCUSSION TITLE
Why and when does VFileInput split my file into a list of bytes?

### DISCUSSION THREAD

**Discussion by bigfooted** [Kitware/trame#255] [Q&A] [answered]
Hi all,
I use a vuetify.VFileInput to read a vtu file. I have a state change in the regular way:
```
@state.change("meshFile")
def load_client_files(meshFile, **kwargs):
```

But large files are chopped up and I end up with a list of bytes instead of one block. I then have to merge this list using 
```
listOfBytes = meshFile.get("content")
reader.SetInputString(b''.join(listOfBytes))
```
I was wondering why this happens, what the file size limit is (a 5 Mb file gets chopped up into 3 pieces for me) and if I can set this limit.  I couldn't find documentation about this.

**Comment by jourdain** [comment, marked as answer]
You should be using the [File helper](https://github.com/Kitware/trame/blob/master/examples/00_howdoi/upload.py) to streamline the chunking.

The chunking is necessary for preventing locking the server network stack and any heartbeat that is used on the WebSocket connection by proxy.

**Reply by bigfooted** [reply]
Thanks! This is exactly what I was looking for!

**Comment by jourdain** [comment]
The chunking is done [here](https://github.com/Kitware/trame-client/blob/master/vue2-app/src/components/Connect/trame/decorators.js#L11) but like I said, it is to prevent network issue with proxy infrastructure.
